### PR TITLE
[RFC]  Internal fields

### PR DESCRIPTION
**Pull request by andrzejewsky**
- [x] I'm confirming that if i made any changes to public APIs or exposed any public APIs they are doccumented.

### CODE REVIEW

**Review by filrak — APPROVED** [review]
I like that but we shouldn't provide that for every comparable, only the ones that have a shared state (so cart, user, locale (?) )

File: packages/core/rfcs/setting-composable-internal-fields.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+```js
+import { useCartFactory } from '@vue-storefront/factories';
+
+export default useCartFactory({ ... })
+```
+
+now usage (example with user):
+
+```js
+import { setCart } from './useCart';
```
Comment (patzick) [inline packages/core/rfcs/setting-composable-internal-fields.md:38]: would need to be like this
```suggestion
import { useCart } from 'composablesPackage';

const {setCart} = useCart()
```

**Review by patzick — APPROVED** [review]
gives more flexibility for sure, but as @filrak mentioned - only makes sense for compostables with shared state :)

File: packages/core/rfcs/setting-composable-internal-fields.md (hunk 1 above)
Comment (filrak) [reply packages/core/rfcs/setting-composable-internal-fields.md:38]: no, that would be an internal api from factory, Andrzej was right

**Pull request merged**